wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1590,13 +1616,22 @@ public void configure(Transport transport) {
                         sshTransport.setSshSessionFactory(sshSessionFactory);
                     }
                 });
+
+                if (nonNull(httpProxyAuth)) {
+                    Authenticator.setDefault(httpProxyAuth);
```

Comment:
> (1) Please, can you explain why we should use httpProxyAuth for ssh connections and httpsProxyAuth for http/https ?

You found an error, and we even don't need setup proxy for ssh.

> (2) I guess there can be some problem with concurrency. I mean you can send two requests with ssh and http links. Then there can be situation when before invoking of command.call() will be invoked Authenticator.setDefault(httpsProxyAuth); for ssh link. Am I wrong?

You are right, but I would prefer don't work around synchronization of invoking authenticator so as it will lead to redundant delays.